docs/orchestrating-elastic-stack-applications/elasticsearch/snapshots.asciidoc

```diff
@@ -285,6 +284,116 @@ PUT /_snapshot/my_s3_repository
 }
 ----
 
+[id="{p}-azure-workload-identity"]
+=== Use Azure Workload Identity
+
+Starting with version 8.16 Elasticsearch supports Azure Workload identity which allows the use of Azure blob storage for Elasticsearch snapshots without exposing Azure credentials directly to Elasticsearch. 
+
+Follow the https://learn.microsoft.com/en-us/azure/aks/workload-identity-deploy-cluster[Azure documentation] for setting up workload identity for the first five steps:
+
+. https://learn.microsoft.com/en-us/azure/aks/workload-identity-deploy-cluster#create-a-resource-group[Create a resource group], if it does not exist yet.
+. https://learn.microsoft.com/en-us/azure/aks/workload-identity-deploy-cluster#create-an-aks-cluster[Create] or https://learn.microsoft.com/en-us/azure/aks/workload-identity-deploy-cluster#update-an-existing-aks-cluster[update] your AKS cluster to enable workload identity.
+. https://learn.microsoft.com/en-us/azure/aks/workload-identity-deploy-cluster#retrieve-the-oidc-issuer-url[Retrieve the OIDC issuer URL].
+. https://learn.microsoft.com/en-us/azure/aks/workload-identity-deploy-cluster#create-a-managed-identity[Create a managed identity] and https://learn.microsoft.com/en-us/azure/aks/workload-identity-deploy-cluster#create-a-kubernetes-service-account[link it to a Kubernetes service account].
+. https://learn.microsoft.com/en-us/azure/aks/workload-identity-deploy-cluster#create-the-federated-identity-credential[Create the federated identity credential].
++
+NOTE: The following steps diverge from the tutorial in the Azure documentation. However, variables initialised as part of the Azure tutorial are still assumed to be present.
++
+. Create an Azure storage account, if it does not exist yet.
++
+[source,sh,subs="attributes,callouts"]
+----
+az storage account create \
+      --name es-storage \
+      --resource-group ${RESOURCE_GROUP} \
+      --location ${LOCATION} \
+      --sku Standard_ZRS \ <1>
+      --encryption-services blob
+----
++
+<1> This can be any of the supported storage account types `Standard_LRS`, `Standard_ZRS`, `Standard_GRS`, `Standard_RAGRS` but not `Premium_LRS` see https://www.elastic.co/guide/en/elasticsearch/reference/current/repository-azure.html[the Elasticsearch documentation] for details.
++
+. Create a container in the storage account, for this example `es-snapshots`.
++
+[source,sh]
+----
+az storage container create \
+   --account-name $STORAGE_ACCOUNT_NAME \
+   --name es-snapshots --auth-mode login
+----
++
+. Create a role assignment between the managed identity and the storage account.
++
+[source,sh,subs="callouts"]
+----
+STORAGE_SCOPE=$(az storage account show \
+  --resource-group $RESOURCE_GROUP \
+  --name $STORAGE_ACCOUNT_NAME --query id -o tsv | sed 's#/##') <1>
+
+az role assignment create \
+  --assignee-object-id $USER_ASSIGNED_IDENTITY_OBJECT_ID \
```

Comment:
Not sure what is `USER_ASSIGNED_IDENTITY_OBJECT_ID`, is it the result of:

```bash
$(az identity show \
    --name "${USER_ASSIGNED_IDENTITY_NAME}" \
    --resource-group "${RESOURCE_GROUP}" \
    --query principalId \
    --output tsv)
```
?

Reply:
Oh good catch: I used a variable from a different Azure tutorial. And you are right it's the result of 
```
export USER_ASSIGNED_IDENTITY_OBJECT_ID="$(az identity show --name "${USER_ASSIGNED_IDENTITY_NAME}" --resource-group "${RESOURCE_GROUP}" --query 'principalId' -otsv)"
```